### PR TITLE
Add Websocket to price service

### PR DESCRIPTION
**Pull request by ali-bahjati**
This adds web socket support for the price service to subscribe to a list of price feeds. It logs their ip and assigns an id for better tracking/debugging. 

I think the client side (on pyth-common-js) is relatively easy, just we should make sure to use a library that supports browser and node at the same time (like `isomorphic-ws`)

### CODE REVIEW

File: third_party/pyth/price-service/src/promClient.ts [hunk 1]
```diff
@@ -31,6 +31,11 @@ export class PromClient {
     buckets: [1, 5, 10, 15, 30, 60, 120, 180],
     labelNames: ["path", "price_id"]
   });
+  private webSocketInteractionCounter = new client.Counter({
+    name: `${SERVICE_PREFIX}websocker_interaction`,
```
Comment (jayantk) [inline third_party/pyth/price-service/src/promClient.ts:35]: typo: websocker

File: third_party/pyth/price-service/src/listen.ts [hunk 2]
```diff
@@ -27,7 +27,9 @@ export type PriceInfo = {
 };
 
 export interface PriceFeedPriceInfo {
-  getLatestPriceInfo(priceFeedId: string): PriceInfo | undefined;
+  getPriceIds(): Set<HexString>;
```
Comment (jayantk) [inline third_party/pyth/price-service/src/listen.ts:30]: nitpick: the name `PriceFeedPriceInfo` is not great. this interface is basically the API for the spy right? maybe something like SpyApi would be better?

**Review by jayantk — APPROVED** [review]
this is really nicely done!

File: third_party/pyth/price-service/src/ws.ts [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+        logger.info(`client ${this.wsId.get(ws)} closed the connection.`);
+        this.promClient?.addWebSocketInteraction("close", "ok");
+
+        this.clientClose(ws);
+      });
+
+      this.promClient?.addWebSocketInteraction("connection", "ok");
+    });
+
+    const pingInterval = setInterval( () => {
```
Comment (Reisen) [inline third_party/pyth/price-service/src/ws.ts:186]: Overall looks great, a quick format with `prettier` or similar run over the files to fix up little spacing issues is the only nitpick.

**Review by Reisen — DISMISSED** [review]

File: third_party/pyth/price-service/src/promClient.ts (hunk 1 above)
Comment (ali-bahjati) [reply third_party/pyth/price-service/src/promClient.ts:35]: Thanks!

**Review by Reisen — APPROVED** [review]

**Pull request merged**